x-pack/plugins/ml/public/embeddables/anomaly_charts/use_anomaly_charts_input_resolver.ts

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useEffect, useMemo, useState } from 'react';
+import { combineLatest, forkJoin, from, Observable, of, Subject } from 'rxjs';
+import { isEqual } from 'lodash';
+import {
+  catchError,
+  debounceTime,
+  distinctUntilChanged,
+  pluck,
+  skipWhile,
+  startWith,
+  switchMap,
+  tap,
+} from 'rxjs/operators';
+import { CoreStart } from 'kibana/public';
+import { TimeBuckets } from '../../application/util/time_buckets';
+import { MlStartDependencies } from '../../plugin';
+import { UI_SETTINGS } from '../../../../../../src/plugins/data/public';
+import {
+  AppStateSelectedCells,
+  ExplorerJob,
+  getSelectionInfluencers,
+  getSelectionJobIds,
+  getSelectionTimeRange,
+  loadDataForCharts,
+} from '../../application/explorer/explorer_utils';
+import { OVERALL_LABEL, SWIMLANE_TYPE } from '../../application/explorer/explorer_constants';
+import { parseInterval } from '../../../common/util/parse_interval';
+import { AnomalyDetectorService } from '../../application/services/anomaly_detector_service';
+import {
+  AnomalyChartsEmbeddableInput,
+  AnomalyChartsEmbeddableOutput,
+  AnomalyChartsServices,
+} from '..';
+import type { CombinedJob } from '../../../common/types/anomaly_detection_jobs';
+import type { ExplorerChartsData } from '../../application/explorer/explorer_charts/explorer_charts_container_service';
+import { processFilters } from '../common/process_filters';
+import { InfluencersFilterQuery } from '../../../common/types/es_client';
+
+const FETCH_RESULTS_DEBOUNCE_MS = 500;
+
+function getJobsObservable(
+  embeddableInput: Observable<AnomalyChartsEmbeddableInput>,
+  anomalyDetectorService: AnomalyDetectorService,
+  setErrorHandler: (e: Error) => void
+) {
+  return embeddableInput.pipe(
+    pluck('jobIds'),
+    distinctUntilChanged(isEqual),
+    switchMap((jobsIds) => anomalyDetectorService.getJobs$(jobsIds)),
+    catchError((e) => {
+      setErrorHandler(e.body ?? e);
+      return of(undefined);
+    })
+  );
+}
```

Comment:
copy from https://github.com/elastic/kibana/blob/32ddd5e795b20d4b6d452412925d1921613d7336/x-pack/plugins/ml/public/embeddables/anomaly_swimlane/swimlane_input_resolver.ts#L48
consider making it a shared function.